### PR TITLE
Resolve unmeet dependencies and remove unused functionality from @cmpsr/contentful-core

### PR DESCRIPTION
**Pull request by guilean**
**Checklist**

- [x] New feature (non-breaking change which adds functionality)

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)

### CODE REVIEW

File: packages/contentful-core/package.json [hunk 1]
```diff
@@ -8,8 +8,7 @@
   "main": "lib/index.js",
   "directories": {
     "lib": "lib",
-    "scripts": "scripts",
-    "test": "__tests__"
+    "scripts": "scripts"
```
Comment (tkanzakic) [inline packages/contentful-core/package.json:11]: is this need it?

File: yarn.lock [hunk 2]
```diff
@@ -20,7 +20,7 @@
     "@jridgewell/gen-mapping" "^0.3.0"
     "@jridgewell/trace-mapping" "^0.3.9"
 
-"@apollo/client@^3.2.2", "@apollo/client@^3.4.8", "@apollo/client@^3.6.2":
+"@apollo/client@^3.4.8", "@apollo/client@^3.6.2":
```
Comment (tkanzakic) [inline yarn.lock:23]: nit: do you know why are we referencing multiple versions?

File: packages/contentful-core/package.json [hunk 3]
```diff
@@ -43,9 +39,7 @@
     "graphql": ">=15"
   },
   "dependencies": {
-    "apollo-link-contentful": "^0.1.3",
-    "cross-fetch": "^3.1.4",
-    "invariant": "^2.2.4"
+    "cross-fetch": "^3.1.4"
```
Comment (tkanzakic) [inline packages/contentful-core/package.json:42]: is this need it?, can it be replace with _global_ fetch?

Comment (tkanzakic) [reply packages/contentful-core/package.json:42]: (commenting in here as I cannot comment below) are dotenv and graphql devDependencies needed?

File: yarn.lock (hunk 2 above)
Comment (guilean) [reply yarn.lock]: `contentful-core` was using a different version than `nextjs-contentful-renderer`. I've updated the version in contentful-core to match

File: packages/contentful-core/package.json (hunk 3 above)
Comment (guilean) [reply packages/contentful-core/package.json]: I've removed `cross-fetch` in favor of default Fetch API used in `HttpLink` when available.
`dontenv` has been removed too

Comment (guilean) [reply packages/contentful-core/package.json]: Regarding `graphql`:

> For version 3.x, graphql is a peer dependency of @apollo/client. This means you should install it separately in your project. The Apollo team made this change to allow developers more control over the version of graphql used in their projects.

**Review by tkanzakic — APPROVED** [review]
🚀

**Pull request merged**